build/docker/zap-baseline.py

```diff
@@ -314,7 +314,8 @@ def main(argv):
                 time.sleep(5)
                 logging.debug('Delay passive scan check ' + str(delay - (datetime.now() - start_scan).seconds) + ' seconds')
 
-        zap_wait_for_passive_scan(zap)
+        # Allow twice as long as the time taken to spider for the passive scanning
+        zap_wait_for_passive_scan(zap, mins * 60 * 2)
```

Comment:
I think it's a fine place to start.

I wonder if we could/should integrate something like this into the core? If we have a way to abandon or terminate a particular plugin (or passive scanning of a request in general) that could be a way to handle things like the passive scanners that use Hirshberg. 
(Like: _Passive scanning a message x took longer than the configured timeout of 120sec, abandoning assessment of passive plugin y._)